libs/utils/utils.js

```diff
@@ -686,3 +686,41 @@ export function loadLana(options = {}) {
   window.addEventListener('error', lanaError);
   window.addEventListener('unhandledrejection', lanaError);
 }
+
+/**
+* Searches for Japanese text in headings and applies a smart word-breaking algorithm by inserting
+* <wbr> between semantic blocks. This allows browsers to break japanese sentences correctly.
+*/
+export async function wordBreakJapanese() {
+  const { miloLibs, codeRoot, locale } = getConfig();
+
+  if (locale.ietf !== 'ja-JP') {
+    return;
+  }
+
+  const base = miloLibs || codeRoot;
+
+  // The thresould value to control word break granularity for long semantic blocks.
+  const WORD_SEG_THRES = 8;
+  const { loadDefaultJapaneseParser } = await import(`${base}/deps/budoux-index-ja.min.js`);
+  const parser = loadDefaultJapaneseParser();
+  document.querySelectorAll('h1, h2, h3, h4, h5').forEach((el) => {
+    // apply budoux to headings
+    parser.applyElement(el, { wordSegThres: WORD_SEG_THRES });
+  });
+
+  const BalancedWordWrapper = (await import(`${base}/deps/./bw2.min.js`)).default;
+  const bw2 = new BalancedWordWrapper();
+  document.querySelectorAll('h1, h2, h3, h4, h5').forEach((el) => {
+    // apply balanced word wrap to headings
+    if (typeof window.requestIdleCallback === 'function') {
```

Comment:
I'd get rid of all this timeout stuff and just call bw2.applyElement(el);
At this point in the load document process you're allowed to call the method.

Reply:
@vhargrave Thank you. We implemented it in Adobe Express not to affect web vitals. However, I didn't see any difference regarding performance when analyzing it with lighthouse. Let me delete it.
https://web.dev/optimize-long-tasks/#manually-defer-code-execution